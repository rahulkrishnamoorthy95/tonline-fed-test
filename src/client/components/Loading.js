import React from 'react';

function Loading({ isLoading }) {

    if (!isLoading) {
        return null;
    }
    
    return (
        <div className="text-center">
            <div className="spinner-border" role="status">
                <span className="sr-only">Loading...</span>
            </div>
        </div>
    );
}

export default Loading;