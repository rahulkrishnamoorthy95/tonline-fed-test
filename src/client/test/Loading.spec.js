import React from 'react';
import Loading from '../components/Loading';

describe('Loading', () => {
	it('renders Loading...', () => {
		const actual = shallow(<Loading />);
		expect(actual.get(0)).toMatchSnapshot();
    });
    
    it('should load based on isLoading', () => {
        const props = {
            isLoading: false
        };
        let wrapper = mount(<Loading {...props} />);    
        expect(wrapper.prop('isLoading')).toEqual(false);

        props.isLoading = true;
        wrapper = mount(<Loading {...props} />);    
        expect(wrapper.prop('isLoading')).toEqual(true);
      });
});