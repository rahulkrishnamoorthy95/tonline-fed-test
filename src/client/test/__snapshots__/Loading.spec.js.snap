// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders Loading... 1`] = `null`;
